### PR TITLE
Large numbers library

### PR DESCRIPTION
**Pull request by kvedala**
#### Description of Change

# New
* library for arbitrarily large integer numbers
* use of it to compute large Fibonacci numbers

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
           "valarray": "cpp",
           "algorithm": "cpp"
      },
-     "C_Cpp.clang_format_style": "{BasedOnStyle: Google, IndentWidth: 4, ColumnLimit: 80, UseTab: Never}",
+     "C_Cpp.clang_format_style": "{ BasedOnStyle: Google, UseTab: Never, IndentWidth: 4, TabWidth: 4,, AllowShortIfStatementsOnASingleLine: false, IndentCaseLabels: true, ColumnLimit: 80, AccessModifierOffset: -4, NamespaceIndentation: All }",
```
Comment (cclauss) [inline .vscode/settings.json:67]:
```suggestion
     "C_Cpp.clang_format_style": "{ BasedOnStyle: Google, UseTab: Never, IndentWidth: 4, TabWidth: 4, AllowShortIfStatementsOnASingleLine: false, IndentCaseLabels: true, ColumnLimit: 80, AccessModifierOffset: -4, NamespaceIndentation: All }",
```

File: DIRECTORY.md [hunk 2]
```diff
@@ -137,8 +137,11 @@
   * [Decimal To Roman Numeral](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/others/Decimal%20to%20Roman%20Numeral.cpp)
   * [Fast Interger Input](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/others/fast_interger_input.cpp)
   * [Fibonacci](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/others/fibonacci.cpp)
+  * [Fibonacci Large](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/others/fibonacci_large.cpp)
```
Comment (cclauss) [inline DIRECTORY.md:140]: These changes happen automatically in GitHub Actions so doing them manually only slows down the review process.

Comment (kvedala) [reply DIRECTORY.md:140]: true, but this was not done manually.

**Comment by kvedala** [comment]
The problem in this pull is with the operator overloading function `+` in `large_number.h`. Any thoughts and suggestions are appreciated.

**Comment by kvedala** [comment]
*note*: CPPLINT passes. the error referred to has been fixed in another pull-request #795

**Comment by kvedala** [comment]
@ayaankhan98

**Comment by kvedala** [comment]
Already included in #805

**Pull request closed without merging**